pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -306,6 +333,33 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 	return nil
 }
 
+// Calculates md5sum of resource-versions of all resources
+// connected to this cluster
+func calcConnectionsHash(con *kdv1.Connections, ns string) string {
+	clusterNames := con.Clusters
+	var buffer bytes.Buffer
+	for _, c := range clusterNames {
+		clusterObj, _ := observer.GetCluster(ns, c)
+		rv := clusterObj.ResourceVersion
```

Comment:
For clusters we might want to use specGenerationToProcess here rather than resourceVersion, as it's a more reliable indicator of whether there was a change that we care about.